Backend/app/routes/brand_dashboard.py

```
@@ -0,0 +1,1627 @@
+    """
+    Add metrics to a campaign
+    """
+    # Validate IDs format
+    validate_uuid_format(campaign_id, "campaign_id")
+    validate_uuid_format(brand_id, "brand_id")
+    
+    try:
+        # Verify campaign belongs to brand
+        campaign_result = supabase.table("sponsorships").select("*").eq("id", campaign_id).eq("brand_id", brand_id).execute()
+        if not campaign_result.data:
+            raise HTTPException(status_code=404, detail="Campaign not found")
```

Ruff (0.14.1)

Abstract `raise` to an inner function

(TRY301)